### ISSUE TITLE
Empty `<subtitle></subtitle>` tags in feeds

### ISSUE THREAD

**Issue by justinmayer** [getpelican/feedgenerator#30]
It seems that since FeedGenerator 1.9.2, some feeds are being rendered with empty `<subtitle></subtitle>` tags in feeds.

Example: https://github.com/getpelican/pelican/pull/2914/commits/99e7af49d11f8fdd57dc42a6efdb24a19835ade1

Presumably this change in behavior was introduced in PR #23. Perhaps we should amend the [conditional check](https://github.com/getpelican/feedgenerator/commit/ae04e8a05b5136eebbc9343137a7e0987737bf50#diff-1e4ba87751d27f77eeae2dae3962c8a75b0a816351b5a86ee27f45dd3eff05a6R338) to ensure that empty subtitle tags are not written to the feed?

cc: @venthur @avaris

**Comment by avaris** [comment]
Well, the comment above those checks does not match the code. If you provide both `subtitle` and `description` you get two `<subtitle>` entries and that's invalid I think (? I don't know spec that well).

**Comment by justinmayer** [comment]
@avaris said:

> If you provide both subtitle and description you get two <subtitle> entries and that's invalid I think

I assume @venthur was under the impression that the second `addQuickElement()` call would overwrite any previous `subtitle` value.

How about replacing said code with the following code? Would it likely address the problem at hand?
```python
# Use subtitle or description if provided, with subtitle taking precedence.
if self.feed['subtitle'] is not None:
    handler.addQuickElement("subtitle", self.feed['subtitle'])
elif self.feed['description'] is not None:
    handler.addQuickElement("subtitle", self.feed['description'])
```

**Comment by avaris** [comment]
That would match the code, but I think the issue is passing empty string (`''`) to those parameters. That passes `is not None` check and ends up as empty `<subtitle>` entry. Probably better to just check the truthiness of those values:

```python
# Use subtitle or description if provided, with subtitle taking precedence.
if self.feed['subtitle']:
    handler.addQuickElement("subtitle", self.feed['subtitle'])
elif self.feed['description']:
    handler.addQuickElement("subtitle", self.feed['description'])
```

**Comment by venthur** [comment]
Sorry for the late reply. I think the fix is rather trivial, this time i added a rather big test that tests against all four possible combinations with the expected result.

**Comment by justinmayer** [comment]
Addressed by #31.
